### PR TITLE
[4.2][cli] cache:clean

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue #37646 .

### Summary of Changes
- added `expired` argument

### Testing Instructions
run `php cli/joomla.php cache:clean`
or
run `php cli/joomla.php cache:clean expired`

check the cache in the admin or cache storage

### Actual result BEFORE applying this Pull Request

cache not cleaned

### Expected result AFTER applying this Pull Request

cache cleaned

### CODE REVIEW

**Comment by brianteeman** [comment]
image

**Comment by brianteeman** [comment]
image

**Comment by alikon** [comment]
fixed

**Comment by brianteeman** [comment]
image

**Comment by alikon** [comment]
i'll check later on a fresh 4.2-dev 
i suppose this happen even without this pr....

**Comment by alikon** [comment]
checked with last 4.2-dev  and it's working fine

File: libraries/src/Console/CleanCacheCommand.php [hunk 1]
```diff
@@ -48,7 +49,30 @@ protected function doExecute(InputInterface $input, OutputInterface $output): in
 
 		$symfonyStyle->title('Cleaning System Cache');
 
-		Factory::getCache()->gc();
+		$cache = Factory::getApplication()->bootComponent('com_cache')->getMVCFactory();
```
Comment (laoneo) [inline libraries/src/Console/CleanCacheCommand.php]: Can you use here `$this->getApplication()`

**Review by laoneo — CHANGES_REQUESTED** [review]

File: libraries/src/Console/CleanCacheCommand.php (hunk 1 above)
Comment (alikon) [reply libraries/src/Console/CleanCacheCommand.php]: done

**Review by laoneo — APPROVED** [review]

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on c59d594bfb79a0b9f66145ba531497612656e624<br /><br />no idea why it didnt work before (although I suspect it was me) but it is working now<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38091">issues.joomla.org/tracker/joomla-cms/38091</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on c59d594bfb79a0b9f66145ba531497612656e624<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38091">issues.joomla.org/tracker/joomla-cms/38091</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38091">issues.joomla.org/tracker/joomla-cms/38091</a>.</sub>

**Pull request merged**